File: BaekJoon/ITBigTech/Big7568/Big.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type Person struct {
	weight int
	height int
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	input, _ := reader.ReadString('\n')
	input = strings.TrimSpace(input)
	N, _ := strconv.Atoi(input)

	people := make([]Person, N)

	for i := 0; i < N; i++ {
		input, _ := reader.ReadString('\n')
		input = strings.TrimSpace(input)
		splits := strings.Split(input, " ")

		weight, _ := strconv.Atoi(splits[0])
		height, _ := strconv.Atoi(splits[1])

		people[i] = Person{weight, height}
	}

	for i := 0; i < N; i++ {
		rank := 1
		for j := 0; j < N; j++ {
			if people[i].height < people[j].height && people[i].weight < people[j].weight {
				rank += 1
			}
		}
		if i == N-1 {
			fmt.Println(rank)
		} else {
			fmt.Print(rank, " ")
		}
	}
}

File: BaekJoon/ITBigTech/Big7568/Big.py
N = int(input())

people = [list(map(int, input().split())) for _ in range(N)]

# 순위 리스트
ranks = [1] * N

for i in range(N):
    rank = 1
    for j in range(N):
        if people[i][0] < people[j][0] and people[i][1] < people[j][1]:
            ranks[i] += 1

print(" ".join(map(str, ranks)))
    

            
